spark/interpreter/src/main/resources/python/zeppelin_ipyspark.py

```diff
@@ -51,3 +51,17 @@
     sqlContext = sqlc = __zSqlc__ = __zSpark__._wrapped
 else:
     sqlContext = sqlc = __zSqlc__ = SQLContext(sparkContext=sc, sqlContext=intp.getSQLContext())
+
+class IPySparkZeppelinContext(PyZeppelinContext):
+
+    def __init__(self, z):
+        super(IPySparkZeppelinContext, self).__init__(z)
+
+    def show(self, obj):
+        from pyspark.sql import DataFrame
+        if isinstance(obj, DataFrame):
+            print(self.z.showData(obj._jdf))
+        else:
+            super(IPySparkZeppelinContext, self).show(obj)
```

Comment:
why not just call `super(IPySparkZeppelinContext, self).show(obj)` always?

Reply:
Here PySparkZeppelinContext enhance and extend PyZeppelinContext so that PySparkZeppelinContext can not only display pandas dataframe but also spark dataframe.

Reply:
Dataset is not supported in PySpark IIUC. There's no dataset related api in pyspark.